Removes metalgen from randomspawns

Time-Green opened the pull request:
## About The Pull Request
Metalgen can't spawn in strange seeds, maintpills, random geysers, egg gland etc anymore

## Why It's Good For The Game

Metalgen is an extremely strong chem, intentionally gated behind both a a rare geyser chem and a randomized recipe. A botanist being able to mass-produce it randomly roundstart throws off the entire balance systems and takes away from the novelty of metalgen synthesis

The randomized recipe system is meant to reward players for putting in extraordinary effort to get an extraordinary reagent. Secret sauce (the only other secret chem) can also not be randomly generated for this reason, I made an oversight by not doing the same for metalgen. 

## Changelog
:cl:
balance: Metalgen cannot spawn as a random chem anymore (strange seeds, maintpills etc)
/:cl:

Time-Green committed:
Removes metalgen from randomspawns

nukularpower commented:
Just thought I'd say, I've done a LOT of strange seed hunting, and today was the first time I've ever found it.   Is it really that big a deal as a very occasional hell yeah?

Time-Green commented:
> Just thought I'd say, I've done a LOT of strange seed hunting, and today was the first time I've ever found it. Is it really that big a deal as a very occasional hell yeah? its not like you can turn the station to plasma if you arent an antag without being in blatant rule violation

I’d at least like to see some effort put in. The odds are if you see metalgen, it’s from botany instead if its intended path. Why ever go through the trouble of working hard for a cool reward if someone else just gets it randomly?? I don’t really care how rare it is, as long as you can get it randomly for basically free it undermines the concept of the super strong funny chem

Bibarra commented:
IIRC metalgen's recipe is still bugged and nonfunctional. (And potentially secret sauce as well) Had a round about a week ago where the recipe sheet had no ingredients listed. Napkin math says that plants rolling metalgen is roughly 1/4000 which potentially makes it rarer and harder to get than finding the recipe.

Time-Green commented:
> IIRC metalgen's recipe is still bugged and nonfunctional. (And potentially secret sauce as well) Had a round about a week ago where the recipe sheet had no ingredients listed. Napkin math says that plants rolling metalgen is roughly 1/4000 which potentially makes it rarer and harder to get than finding the recipe.

How is harder? You just get it randomly for doing your job

HoodMilk commented:
Who is complaining about this? Even in its currently state metalgen is extremely rare and mostly a novelty chemical used to turn rooms gold for laughs. It can be used for griefing with plasma yeah but why the hell would anybody go through the trouble when you can sign on as atmos tech and burn the entire station down roundstart. It's fun, keep it in.

the-orange-cow commented:
> > IIRC metalgen's recipe is still bugged and nonfunctional. (And potentially secret sauce as well) Had a round about a week ago where the recipe sheet had no ingredients listed. Napkin math says that plants rolling metalgen is roughly 1/4000 which potentially makes it rarer and harder to get than finding the recipe.
> 
> How is it harder? You just get it randomly for doing your job

For what its worth, it isn't that simple.  You need specific dedicated setups to farm random chems.  It isn't something that will happen every round, or even every five rounds.

MrMelbert commented:
This is a little disappointing because I think Gaseous Decomposition + Metalgen plants are far and above the *coolest* not-antag thing you can make it Botany and it's so rare I've only done it twice in 600 hours of play. 

> I’d at least like to see some effort put in. The odds are if you see metalgen, it’s from botany instead if its intended path. 

Would you rather see Metalgen not at all, or rarely?

MercuryArrow commented:
I'm inclined to agree with Melbert and Barra on this - not only is metalgen currently bugged/broken across multiple servers, but due to how secret recipes can differ across servers, they're one of the only things I've ever seen that can be broken on Sybil and Manuel and working fine on Terry, for example.  It may genuinely be no exaggeration to say that across the course of its inclusion in the codebase, metalgen recipe generation or metalgen itself have been busted for more time than they've been functioning as intended. We should absolutely not be gutting fairly rare/low chance alternatives to acquire the chemical when the 'legitimate' way breaks for months at a time in a wildly unpredictable fashion and has never been truly satisfactorily solved.

Putting that whole headache aside, the balance implications and capability metalgen has for affecting a round tend to be overstated - there are like, two things you can do with metalgen that are truly impactful and can't be easily replicated by other avenues, and those things mostly involve burning down the station in a slightly more grandiose way than if you'd done something that would have taken half an hour less not counting the multiple rounds you spent digging up the recipe to begin with.  If material datums actually reliably modified weapon force and other properties like the code implies they're meant to, it would be a much bigger deal, but this hasn't ever really taken shape.

CMDR-Gungnir commented:
I agree with Bib, Melbert and Mercury, but also want to bring up a couple points of my own.

>and takes away from the novelty of metalgen synthesis
I dunno, it's pretty rare that it happens and quite frankly a Botanist throwing a flower that explodes into a room of gold has a pretty high novelty value, especially given I've yet to ever see it.

>How is it harder? You just get it randomly for doing your job
Excluding "Rolling it as a random mutation" because the only way you're going to really even see it show up in there is either Extreme Fucking Luck, or you're completely neglecting doing your job properly. Because contrary to popular belief, you don't want to be rolling random chems in the plants you're growing to feed people. To even hope to see it """regularly""" (this would still be Rare) you'd have to dedicate your entire setup to random rolling. But you're also not changing that if the PR body is to be believed.

Which means you're talking solely about Strange Seeds. Which are not a Botanist's job, it's a novelty the job can do. And even then, it's completely up to Luck, on a very low chance to happen. So "randomly for doing your job" isn't going to happen. I've had rounds where the chefs don't ask much of me and I just roll a bunch of plants and I've never had Metalgen pop up in my plants, yet. Even neglecting your job doesn't give it to you often.

>I’d at least like to see some effort put in. The odds are if you see metalgen, it’s from botany instead if its intended path.
No, usually it's come out of chemistry, I've still yet to see a Botanist do it.

ExcessiveUseOfCobblestone commented:
Im on team timegreen

Metalgen should retain its rarity and if you want to add stuff worthy of getting it then just do that instead of letting it slip down the slope (albeit the very small inclined slope considering the chance of appearing). The whole point of the chem is to be locked behind this recipe, it loses its novelty if its showing up through other means regardless of how infrequent they are (especially considering if thats the only way it's practically appearing).

If a portion of the metalgen use is cool then just have it replicated in another form without it being metalgen. We did it with say bronze/brass armor or the glitter. If its only cool because its metalgen then see above.

the recipe being broken does not make it ok to just make it available elsewhere when the obvious solution is to just fix/change the recipe to not be broken.

MrMelbert commented:
what if we removed metalgen from the intended path instead and replaced it with something that warrants being nigh impossible to create

the-orange-cow commented:
> what if we removed metalgen from the intended path instead and replaced it with something that warrants being nigh impossible to create

Admindrazine recipe gogogo!

Time-Green commented:
If metalgen recipes are broken on a lot of servers, than that just needs to be fixed. I'm not letting balance decisions be influenced by a bug that can be fixed (I haven't seen it break, but I'll look into it somewhere in the next two weeks)

> what if we removed metalgen from the intended path instead and replaced it with something that warrants being nigh impossible to create

Replacing metalgen with a stronger chem would be fine with me. There need to be some turbo chems to tickle brains that can only be gotten through secret recipes, otherwise what's the allure in making an impossible chem if you can get it randomly? Secret sauce (the only other secret reagent) can also not be randomly generated for that reason, it was an oversight on my part by not doing the same for metalgen when I added it (or rebalanced it, it's been through a lot)

Time-Green commented:
I checked out of curiosity and it seems metalgen recipe generation is indeed broken 50/50 of the time. Probably something in the recipe generation died. 
- Manuel: No metalgen
- Campbel: Metalgen
- Basil: Metalgen
- Sybil: No metalgen
- Terry: Metalgen

This has no influence on my decision, it's just a bug that I will need to fix. This does remind me that it's probably time to whitelist the secret recipe system to make sure that it gets obtainable reagents only, not random medicine metabolites that you can only get through mass gibbing, but I'll have a lot of free time in a few weeks

chizzy376 commented:
Removing this chem for the sake of "but what if players abuse it" means nothing, since using it for bad as nontot is griefing-- a blatant rule violation. I'd like to see this pr reverted, given that this just removes the emergent gameplay.
